traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/util/LanguidState.java

```diff
@@ -83,8 +83,8 @@ private void initPorts(final JsonNode routerJson) {
 			trafficRouterManager.setApiPort(apiPort);
 		}
 
-		if (routerJson.has("secure.port")) {
-			setSecurePort(routerJson.get("secure.port").asInt());
+		if (routerJson.has("httpsPort")) {
```

Comment:
I believe this needs to use `hasNonNull` instead of `has`; otherwise, L87 could cause a null pointer exception for TRs that don't have `httpsPort` set (and hence have a `null` value). In fact, just for extra safety/consistency, we should probably change all of the `has` checks into `hasNonNull` in this method.

Reply:
You are totally correct on the use of hasNonNull. I changed line 86 to use the correct method. I did not change the other lines.

If for some reason someone does not put an http port, the field is not showed in the CrConfig.json. Thus, the if statement will evaluate false, and no exception will be triggered (at least not because of that line).

And if for the very odd reason someone creates the api.port and/or secure.api.port, assigns them to the CCR profile, but leave them as empty value. The CrConfig.json will show them as empty strings `secure.api.port: ""`. I did the test and TR seems to simply ignore them, and uses the default ones 3333/3443.